apps/ui/src/Pages/Messages/MessageCard.tsx

```diff
@@ -0,0 +1,101 @@
+import { Message, MessageType } from '@communecar/types';
+import {
+  Avatar,
+  Box,
+  Button,
+  CircularProgress,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  Typography,
+} from '@mui/material';
+import { formatDateRelative } from '../../utils/format/formatDateRelative';
+import { RequestActions } from '../../types/actions';
+import { useRespondToMessage } from '../../hooks/Messages/useRespondToMessage';
+import { actionTextDisplay } from '../../utils/messages/actionTextDisplay';
+
+export interface MessageCardProps {
+  message: Message;
+  onActionComplete: (messageId: string, success: boolean) => void;
+}
+
+const MessageCard: React.FC<MessageCardProps> = ({
+  message,
+  onActionComplete,
+}) => {
+  const { submitRespondToMessage, isLoading, error } = useRespondToMessage();
+
+  const handleAction = async (action: RequestActions) => {
+    const success = await submitRespondToMessage(message.id, action);
+    if (success && !isLoading) {
+      alert(`Action ${action} was successful!`);
+      onActionComplete(message.id, true);
+    } else {
+      alert(`Failed to ${action} the request: ${error?.message}`);
```

Comment:
change this to use toasts or snackbars

Reply:
we talk about this - we will do this to the whole app in a diffrent branch